swap/htlc.go

```diff
@@ -25,9 +26,51 @@ const (
 	HtlcNP2WSH
 )
 
+// ScriptVersion defines the HTLC script version.
+type ScriptVersion uint8
+
+const (
+	// HtlcV1 refers to the original version of the HTLC script.
+	HtlcV1 ScriptVersion = iota
+
+	// HtlcV2 refers to the improved version of the HTLC script.
+	HtlcV2
+)
+
+// htlcScript defines an interface for the different HTLC implementations.
+type HtlcScript interface {
+	// genSuccessWitness returns the success script to spend this htlc with
+	// the preimage.
+	genSuccessWitness(receiverSig []byte, preimage lntypes.Preimage) wire.TxWitness
+
+	// genTimeoutWitness returns the timeout script to spend this htlc after
+	// timeout.
+	GenTimeoutWitness(senderKey [33]byte, senderSig []byte) wire.TxWitness
+
+	// isSuccessWitness checks whether the given stack is valid for
+	// redeeming the htlc.
+	IsSuccessWitness(witness wire.TxWitness) bool
+
+	// getScript returns the htlc script.
+	GetScript() []byte
+
+	// maxSuccessWitnessSize returns the maximum witness size for the
+	// success case witness.
+	MaxSuccessWitnessSize() int
+
+	// maxTimeoutWitnessSize returns the maximum witness size for the
+	// timeout case witness.
+	MaxTimeoutWitnessSize() int
+
+	// sequence returns the sequence to spend this htlc.
+	Sequence() uint32
```

Comment:
Make more specific like `SuccessSequence`? So that it is clear that this isn't needed for the timeout path

Reply:
done